src/lib/functions/server.ts

```diff
@@ -157,7 +157,7 @@ export const createHandler = function (options) {
       'client-ip': [remoteAddress],
       'x-nf-client-connection-ip': [remoteAddress],
       'x-nf-account-id': [options.accountId],
-      'x-nf-site-id': [options?.siteInfo?.id] ?? 'unlinked',
```

Comment:
This was the problem. `[options?.siteInfo?.id]` is never falsy, even when `options?.siteInfo?.id` is `undefined`, so the fallback string never kicked in.